task_processing/plugins/mesos/execution_framework.py

```diff
@@ -481,110 +267,95 @@ def create_new_docker_task(self, offer, task_config):
 
         return task
 
-    def stop(self):
-        pass
-
     ####################################################################
     #                   Mesos driver hooks go here                     #
     ####################################################################
 
     def slaveLost(self, drive, slaveId):
-        log.error("Slave lost: {id}".format(id=str(slaveId)))
+        log.warning("Slave lost: {id}".format(id=str(slaveId)))
 
     def registered(self, driver, frameworkId, masterInfo):
-        self.framework_id = frameworkId.value
-        paasta_print("Registered with framework ID %s" % frameworkId.value)
+        log.info("Registered with framework ID {id}".format(id=frameworkId.value))
 
-        self.reconcile_start_time = time.time()
-        driver.reconcileTasks([])
+    def reregistered(self, driver, frameworkId, masterInfo):
+        log.warning("Registered with framework ID {id}".format(id=frameworkId.value))
 
     def resourceOffers(self, driver, offers):
-        if self.frozen:
-            return
+        if self.driver is None:
+            self.driver = driver
 
-        if self.within_reconcile_backoff():
-            paasta_print(
-                'Declining all offers since we started reconciliation '
-                'too recently'
-            )
-            for offer in offers:
-                driver.declineOffer(offer.id)
-        else:
-            for idx, offer in enumerate(offers):
-                if offer.slave_id.value in self.blacklisted_slaves:
-                    log.critical(
-                        'Ignoring offer {0} from blacklisted slave {1}'.format(
-                            offer.id.value, offer.slave_id.value
-                        )
+        for offer in offers:
+            if self.task_queue.empty():
+                log.info("Declining offer {id} because there are no more tasks to launch.".format(id=offer.id.value))
+                driver.declineOffer(offer.id, self.offer_decline_filter)
+                continue
```

Comment:
Done! I am a little worried about the impact of suppressing and reviving offers on DRF algorithm of mesos. But, let's try it out.